libraries/ESP8266mDNS/src/LEAmDNS_Control.cpp

```diff
@@ -79,10 +79,8 @@ bool MDNSResponder::_process(bool p_bUserContext) {
         }
     }
     else {
-        bResult = ((WiFi.isConnected() ||               // Either station is connected
-                    WiFi.softAPgetStationNum()>0) &&    // Or AP has stations connected
-                   (_updateProbeStatus()) &&            // Probing
-                   (_checkServiceQueryCache()));        // Service query cache check
+        bResult = _updateProbeStatus() &&           // Probing
```

Comment:
The first part of the condition in the current code is supposed to mean "there's somebody to send to", i. e. : if the station is connected to an ap, then there's at least one receiver OR if there's someone connected to the softap there's at least one receiver.
i think that removing that part of the condition makes MDNS operate needlessly.

Reply:
It indeed does make mDNS operate even when there is no connection.
However mDNS is now supposed to be interface agnostic, bound to an IPAddress defining an interface instead of trying to find out which interface is up or to answer to. The only place where AP or STA should be dealt with is when the IP address from `begin()` is not set and a default interface has to be selected to keep with backward compatibility.

The first test `WiFi.isConnected()` checks wether STA is up, the second one counts the number of WiFi clients connected to the AP without really checking if AP is up. 

I propose to merge this PR asap so one can debug #5866 (@LaborEtArs and @hreintke will hopefully help on this) (and UDP is fixed), and I will propose a new function to check on interface status with an IP address passed as argument because we lack&need this functionality.